src/ln/channel.rs

```diff
@@ -1820,13 +1862,48 @@ impl Channel {
 						update_fulfill_htlcs: Vec::new(),
 						update_fail_htlcs,
 						update_fail_malformed_htlcs,
+						update_fee: None,
 						commitment_signed
 					}), to_forward_infos, revoked_htlcs, monitor_update))
 				} else {
 					Ok((None, to_forward_infos, revoked_htlcs, self.channel_monitor.clone()))
 				}
 			}
 		}
+
+	}
+
+	pub fn send_update_fee(&mut self, feerate_per_kw: u64) -> Result<Option<msgs::UpdateFee>, HandleError> {
```

Comment:
nit: dont make this pub unless its actually used somewhere in ChannelManager (and document the "cannot be called twice without a send_commitment in between" invariant).